File: apps/demojava/app/src/main/java/com/tradehousemedia/tradecore/java/examples/TradecoreAdView320x50Activity.java
package com.tradehousemedia.tradecore.java.examples;

import android.os.Bundle;
import android.util.Log;
import androidx.annotation.Nullable;
import com.tradehousemedia.tradecore.core.TradecoreAd;
import com.tradehousemedia.tradecore.core.TradecoreAdView;
import com.tradehousemedia.tradecore.core.TradecoreError;
import com.tradehousemedia.tradecore.core.listeners.TradecoreAdViewListener;
import com.tradehousemedia.tradecore.java.R;
import com.tradehousemedia.tradecore.java.utils.BaseExampleActivity;
import org.jetbrains.annotations.NotNull;

public class TradecoreAdView320x50Activity extends BaseExampleActivity {

    private static final String TAG = "ExampleActivity";

    private TradecoreAdView adView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_example);
        setTitle();
        getRefreshButton().setOnClickListener(v -> loadAd());

        loadAd();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        adView.destroy();
    }

    private void loadAd() {
        getAdContainer().removeAllViews();

        adView = new TradecoreAdView(this);
        adView.setZoneId("tradecore-zone-3025");
        adView.setAdListener(createListener());
        adView.loadAd();
        getAdContainer().addView(adView);
    }

    private TradecoreAdViewListener createListener() {
        return new TradecoreAdViewListener() {
            @Override
            public void onAdLoaded(@NotNull TradecoreAd tradecoreAd) {

            }

            @Override
            public void onAdFailed(@NotNull TradecoreError tradecoreError) {
                Log.d(TAG, "Ad failed to load: " + tradecoreError.getMessage());
            }

            @Override
            public void onAdImpression(@NotNull TradecoreAd tradecoreAd) {

            }

            @Override
            public void onAdClicked(@NotNull TradecoreAd tradecoreAd) {

            }

            @Override
            public void onAdClosed(@NotNull TradecoreAd tradecoreAd) {

            }

            @Override
            public void onAdOpened(@NotNull TradecoreAd tradecoreAd) {

            }
        };
    }

}

File: apps/demokotlin/app/src/main/java/com/tradehousemedia/tradecore/kotlin/examples/TradecoreAdView320x50Activity.kt
package com.tradehousemedia.tradecore.kotlin.examples

import android.os.Bundle
import android.util.Log
import android.view.View
import com.tradehousemedia.tradecore.core.TradecoreAdView
import com.tradehousemedia.tradecore.core.TradecoreError
import com.tradehousemedia.tradecore.core.listeners.TradecoreAdViewListener
import com.tradehousemedia.tradecore.kotlin.R
import com.tradehousemedia.tradecore.kotlin.utils.BaseExampleActivity

class TradecoreAdView320x50Activity : BaseExampleActivity() {

    companion object {
        private const val TAG = "ExampleActivity"
    }

    private var adView: TradecoreAdView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_example)
        setTitle()
        refreshButton.setOnClickListener { v: View? -> loadAd() }

        loadAd()
    }

    override fun onDestroy() {
        super.onDestroy()
        adView?.destroy()
    }

    private fun loadAd() {
        adContainer.removeAllViews()

        val adView = TradecoreAdView(this).apply { adView = this }
        adView.zoneId = "tradecore-zone-3025"
        adView.setAdListener(createListener())
        adView.loadAd()
        adContainer.addView(adView)
    }

    private fun createListener(): TradecoreAdViewListener {
        return object : TradecoreAdViewListener {
            override fun onAdFailed(tradecoreError: TradecoreError) {
                Log.d(TAG, "Ad failed to load: " + tradecoreError.message)
            }
        }
    }

}
